### PR TITLE
Bug: apply final_hidden_states*=self.routed_scaling_factor at MoE lay…

### PR DESCRIPTION
**Pull request by byjiang1996**
…er if epmoe is enabled

## Motivation

Issue: https://github.com/sgl-project/sglang/issues/8402

Also noticed significant regression when running epmoe during recent GLM4.5 support work: GSM8K accuracy drops from 0.965 to 0.745 when EPMOE is enabled. Accuracy is good for TP & DeepEP.

```
python3 -m sglang.launch_server --model  /shared/public/elr-models/zai-org/GLM-4.5 --tp-size 8 --trust-remote-code

python3 benchmark/gsm8k/bench_sglang.py
```

## Modifications

There is a bug in DeepSeek-V2 and GLM-4.5 related to how routed_scaling_factor is applied in MoE (Mixture-of-Experts) layers.

Currently, the routed_scaling_factor is applied in three different places, leading to ambiguity:
- self.topk.forward: The scaling is applied only in the n out of m condition, but not in the m - n out of m condition.
- self.experts.forward: Same as above — the factor is only applied in n out of m.
- Model-level logic: The model itself may apply routed_scaling_factor, but it cannot know whether self.topk or self.experts have already done so.

This results in uncertain and inconsistent scaling, as the model layer has no visibility into whether routed_scaling_factor has already been applied upstream.

🎪 TL;DR: The model can't know if it should apply * routed_scaling_factor or not, because topk and experts may or may not have already done it, depending on the code path.

My PR forces model layer to apply * routed_scaling_factor when EPMOE is enabled because from the current codebase, epmoe won't apply *routed_scaling_factor by itself. 

Need to follow up and possibly refactor sglang moe codebase to make it clear which layer should apply * routed_scaling_factor 

## Accuracy Test
- GLM4.5 GSM8K accuracy jumps from 0.745 to 0.965 under EPMOE which is the same as TP & DeepEP results
- Need to run gsm8k for deepseek-v3 too - WIP
## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [x] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [x] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [x] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [x] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [x] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/models/deepseek_v2.py [hunk 1]
```diff
@@ -506,8 +506,11 @@ def forward_normal(
         else:
             kwargs["router_logits"] = router_logits
         final_hidden_states = self.experts(**kwargs)
-        if not _is_cuda and not _use_aiter:
+        if not _is_cuda and not _use_aiter or global_server_args_dict["enable_ep_moe"]:
```
Comment (trevor-m) [inline python/sglang/srt/models/deepseek_v2.py]: Maybe we can have a property in `self.experts` whether to apply routing scaling factor in the model?
Or can we fuse it for EpMoe too?

These are the scenarios I know about for cuda:
* FusedMoE triton backend: multiply is fused into `moe_sum_reduce`, but also need to divide from shared experts in `biased_grouped_topk` to cancel it out
* FusedMoE model_opt FP4 (with or without enable_ep_moe): applied in `ModelOptNvFp4FusedMoEMethod` but https://github.com/sgl-project/sglang/pull/8364 will fuse multiply into `biased_grouped_topk`. I'm worried this change will cause it to be applied twice for this path
* EpMoE - was missing but this PR will fix and bring to model forward
* DeepEpMoE - in model forward

Comment (ch-wan) [reply python/sglang/srt/models/deepseek_v2.py]: Let me fix this after 8515 is merged.

**Review by trevor-m — APPROVED** [review]
